tests/pipelines/test_pipelines_zero_shot_object_detection.py

```diff
@@ -0,0 +1,283 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from transformers import MODEL_FOR_ZERO_SHOT_OBJECT_DETECTION_MAPPING, is_vision_available, pipeline
+from transformers.testing_utils import (
+    is_pipeline_test,
+    nested_simplify,
+    require_tf,
+    require_torch,
+    require_vision,
+    slow,
+)
+
+from .test_pipelines_common import ANY, PipelineTestCaseMeta
+
+
+if is_vision_available():
+    from PIL import Image
+else:
+
+    class Image:
+        @staticmethod
+        def open(*args, **kwargs):
+            pass
+
+
+@require_vision
+@require_torch
+@is_pipeline_test
+class ZeroShotObjectDetectionPipelineTests(unittest.TestCase, metaclass=PipelineTestCaseMeta):
+
+    model_mapping = MODEL_FOR_ZERO_SHOT_OBJECT_DETECTION_MAPPING
+
+    def get_test_pipeline(self, model, tokenizer, feature_extractor):
+        object_detector = pipeline(
+            "zero-shot-object-detection", model="hf-internal-testing/tiny-random-owlvit-object-detection"
+        )
+
+        examples = [
+            {
+                "images": "./tests/fixtures/tests_samples/COCO/000000039769.png",
+                "text_queries": ["cat", "remote", "couch"],
+            }
+        ]
+        return object_detector, examples
+
+    def run_pipeline_test(self, object_detector, examples):
+        batch_outputs = object_detector(
+            examples,
+            threshold=0.0,
+        )
+
+        self.assertEqual(len(examples), len(batch_outputs))
+        for outputs in batch_outputs:
+            for output_per_image in outputs:
+                scores = output_per_image["scores"]
+                labels = output_per_image["labels"]
+                boxes = output_per_image["boxes"]
+
+                assert len(scores) == len(boxes) and len(boxes) == len(labels)
+
+                for score in scores:
+                    self.assertEqual(score, ANY(float))
+                for label in labels:
+                    self.assertEqual(label, ANY(str))
+                for box in boxes:
+                    self.assertEqual(box, {"xmin": ANY(int), "ymin": ANY(int), "xmax": ANY(int), "ymax": ANY(int)})
+
+    @require_tf
+    @unittest.skip("Zero Shot Object Detection not implemented in TF")
+    def test_small_model_tf(self):
+        pass
+
+    @require_torch
+    def test_small_model_pt(self):
+        object_detector = pipeline(
+            "zero-shot-object-detection", model="hf-internal-testing/tiny-random-owlvit-object-detection"
+        )
+
+        outputs = object_detector(
+            "./tests/fixtures/tests_samples/COCO/000000039769.png",
+            text_queries=["cat", "remote", "couch"],
+            threshold=0.64,
+        )
+
+        self.assertEqual(
+            nested_simplify(outputs, decimals=4),
+            [
+                {
+                    "scores": [0.6748, 0.6456, 0.7235, 0.642],
```

Comment:
I feel like these outputs are unnatural.

Why not `[{"score": 0.67, "label": "remote", "bbox": ...}, {...}, ]` instead ?
The problem with this output is that most users will probably have to zip anyway, and there's room for error if the list sizes don't match.

Reply:
Also why don't we sort by descending score ?

And instead of `threshold` which is likely to be model dependant, why don't we use `top_k` instead ?

(We can keep threshold, but `top_k`  feels more usable to me).

Reply:
I slightly have a different view here. Consider this text_query = ["human-face", "car"] and the image is a street setting. If cars are pretty easily detectable, they will have high scores and dominate the top_k detections. Human Faces in streets being harder to detect will always have a low confidence. Hence playing around with the threshold would be easier as compared to top_k.

@alaradirik  any thoughts?